### ISSUE TITLE
vue/return-in-computed-property: False positive on exhaustive switch

### ISSUE THREAD

**Issue by ilyub** [vuejs/eslint-plugin-vue#1727]
**Tell us about your environment**

- **ESLint version:** 8.3.0
- **eslint-plugin-vue version:** 8.1.1
- **Node version:** 14.18.0
- **Operating System:** Windows 10

**Please show your full configuration:**

```json5
parser: "vue-eslint-parser",
  parserOptions: {
    ecmaFeatures: {
      jsx: true
    },
    ecmaVersion: 2017,
    extraFileExtensions: [".vue"],
    parser: "@typescript-eslint/parser",
    project: "./tsconfig.json",
    sourceType: "module"
  },
rules: {
  "vue/return-in-computed-property": "warn"
}
```

**What did you do?**	
```vue
setup() {
  let x: "a" | "b";
  return {
      f: computed((): number => {
        switch (x) {
          case "a":
            return 1;

          case "b":
            return 2;
        }
      })
  }
}
```

**What did you expect to happen?**

This switch is exhaustive. So I expect no warning.

**What actually happened?**

Getting "Expected to return a value in computed function"

**Comment by ota-meshi** [comment]
Thank you for posting issue.

The `vue/return-in-computed-property` rule uses ESLint's **Code Path Analysis**, so it's impossible to check the TypeScript types.

https://eslint.org/docs/developer-guide/code-path-analysis

I think you can turn off the `vue/return-in-computed-property` rule because your code is checked by TypeScript.

Thank you for your understanding.
